cmd/pitr/collector/collector.go

```diff
@@ -122,33 +128,64 @@ func (c *Collector) close() error {
 	return c.db.Close()
 }
 
+func (c *Collector) filterBinLogs(logs []pxc.Binlog) ([]pxc.Binlog, error) {
+	result := make([]pxc.Binlog, 0)
+	for _, v := range logs {
+		set, err := c.db.GetGTIDSet(v.Name)
```

Comment:
`get_gtid_set_by_binlog` is potentially expensive operation, we should call `GetGTIDSet` only if needed